File: in_Java/0189 Rotate Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k % nums.length;
        rev(nums, 0, nums.length-1);
        rev(nums, 0, k-1);
        rev(nums, k, nums.length-1);
    }
    private void rev(int[] nums, int i, int j){
        int t;
        while(i < j){
            t = nums[i];
            nums[i] = nums[j];
            nums[j] = t;
            i++;
            j--;
        }
    }
}

File: in_Python/0189 Rotate Array.py
class Solution(object):
    def rev(self, arr, i, j):
        while i < j:
            t = arr[i]
            arr[i] = arr[j]
            arr[j] = t
            i += 1
            j -= 1
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        k = k % len(nums)
        self.rev(nums,0,len(nums)-k-1)
        self.rev(nums,len(nums)-k,len(nums)-1)
        self.rev(nums,0,len(nums)-1)
